### PR TITLE
OCD-2973 - Revise list of test tools to select for a 2015 Edition listing

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

**Review by andlar — COMMENTED** [review]
I only see changes to the running database. Don't we need, at least, an addition to the criterion_attribute table in the dev/data-model.sql file? As well as the new mapping table?

File: changes/ocd-2973.sql [hunk 1]
```diff
@@ -0,0 +1,1396 @@
...
+ALTER TABLE openchpl.certification_criterion_attribute DROP CONSTRAINT criterion_unique;
+ALTER TABLE openchpl.certification_criterion_attribute ADD CONSTRAINT criterion_unique UNIQUE (criterion_id);
+
+-- Add new column to indicate whether the criterion supports test tools
+ALTER TABLE openchpl.certification_criterion_attribute ADD COLUMN IF NOT EXISTS test_tool BOOL NOT NULL DEFAULT FALSE;
+
+-- Add the new test tools
+INSERT INTO openchpl.test_tool
+(name, last_modified_user)
+SELECT 'NCPDP Electronic Prescribing (eRx) Testing Tool', 1
```
Comment (kekey1) [inline changes/ocd-2973.sql]: Is the last_modified_user the value you intended? I was thinking we normally use -1.

File: changes/ocd-2973.sql [hunk 2]
```diff
@@ -0,0 +1,1396 @@
...
+WHERE NOT EXISTS
+	(SELECT *
+	FROM openchpl.test_tool
+	WHERE name = 'NCPDP Electronic Prescribing (eRx) Testing Tool');
+
+INSERT INTO openchpl.test_tool
+(name, last_modified_user)
+SELECT 'NIST General Validation Tool (GVT)', 1
```
Comment (kekey1) [inline changes/ocd-2973.sql]: Same thing with 1 vs -1

File: changes/ocd-2973.sql [hunk 3]
```diff
@@ -0,0 +1,1396 @@
...
+	FROM openchpl.test_tool
+	WHERE name = 'NIST General Validation Tool (GVT)');
+
+-- INSERT or UPDATE (also called UPSERT - learned something new) the criterion and whther it supports test tools
+INSERT INTO openchpl.certification_criterion_attribute (criterion_id, test_tool, last_modified_user)
+SELECT certification_criterion_id, false, -1
+FROM openchpl.certification_criterion
+WHERE number = '170.314 (a)(1)'
+ON CONFLICT ON CONSTRAINT criterion_unique DO UPDATE SET
+	test_tool = EXCLUDED.test_tool;
```
Comment (kekey1) [inline changes/ocd-2973.sql:48]: Just curious - what is EXCLUDED?

**Review by kekey1 — CHANGES_REQUESTED** [review]

**Review by kekey1 — CHANGES_REQUESTED** [review]
Agree with Andrew's comment about needing some changes in data-model.sql

File: dev/openchpl_data-model.sql [hunk 4]
```diff
@@ -3435,6 +3437,23 @@ CREATE TABLE openchpl.listing_cures_status_statistic (
 );
 CREATE INDEX idx_listing_cures_status_stat_date on openchpl.listing_cures_status_statistic (statistic_date);
 
+CREATE TABLE openchpl.listing_cures_status_statistic_by_acb (
```
Comment (andlar) [inline dev/openchpl_data-model.sql]: I don't think this table should be here

File: changes/ocd-2973.sql [hunk 5]
```diff
@@ -0,0 +1,1408 @@
+INSERT INTO openchpl.deprecated_api
+(http_method, api_operation, change_description, last_modified_user)
+SELECT 'GET', 
+	'/data/test_tools',
+	'This endpoint is deprecated and will be removed in a future release. The test_tools are now dependent on the criteria they are associted with and can be found in the CertifiedProductSearchDetails.',
```
Comment (andlar) [inline changes/ocd-2973.sql]:
```suggestion
	'This endpoint is deprecated and will be removed in a future release. The test_tools are now dependent on the criteria they are associated with and can be found in the CertifiedProductSearchDetails.',
```

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Pull request merged**